src/main/java/com/google/sps/data/DeliverySystem.java

```diff
@@ -26,16 +26,74 @@
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
+import java.util.List;
 
 /** 
  * Class that assigns orders and creates delivery journeys for delivery slots.
  */
 public class DeliverySystem {
+  private final Integer MAX_WAYPOINTS = 25;
+  private PathFinder pathFinder;
+
+  public DeliverySystem(PathFinder pathFinder) {
+    this.pathFinder = pathFinder;
+  }
+
   /**
    * Finds the order in which points must be visited such that the journey lasts minimum time and it
    * delivers all orders assigned in findOrdersForDeliveryRequest.
    */
-  public ArrayList<Point> findOptimalDeliveryJourneyForDeliverySlot(DeliverySlot deliverySlot) throws ApiException, IOException, InterruptedException {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public ArrayList<Point> findOptimalDeliveryJourneyForDeliverySlot(DeliverySlot deliverySlot) throws ApiException, BadRequestException, DataNotFoundException, IOException, InterruptedException {
+    Journey journey = findOrdersForDeliveryRequest(deliverySlot);
+    return journey.findOptimalOrderForWaypoints();
+  }
+
+  /**
+   * Assigns a set of orders for the delivery request such that they fit the [startTime, endTime]
+   * timeslot.
+   */
+  public Journey findOrdersForDeliveryRequest(DeliverySlot deliverySlot)  throws ApiException, BadRequestException, DataNotFoundException, IOException, InterruptedException {
```

Comment:
nit: This method is called findOrders, but it actually constructs a journey, this is confusing. Call it `createJourneyForDeliverySlot` if that's what it does.